account_cutoff_base/models/account_cutoff.py

```diff
@@ -0,0 +1,391 @@
+        related='parent_id.company_currency_id',
+        string="Company Currency", readonly=True)
+    partner_id = fields.Many2one(
+        'res.partner', string='Partner', readonly=True)
+    account_id = fields.Many2one(
+        'account.account', 'Account',
+        domain=[('deprecated', '=', False)], required=True, readonly=True)
+    cutoff_account_id = fields.Many2one(
+        'account.account', string='Cut-off Account',
+        domain=[('deprecated', '=', False)], required=True, readonly=True)
+    cutoff_account_code = fields.Char(
+        related='cutoff_account_id.code',
+        string='Cut-off Account Code', readonly=True)
+    analytic_account_id = fields.Many2one(
+        'account.analytic.account', string='Analytic Account',
+        domain=[('account_type', '!=', 'closed')], readonly=True)
+    analytic_account_code = fields.Char(
+        related='analytic_account_id.code',
+        string='Analytic Account Code', readonly=True)
+    currency_id = fields.Many2one(
+        'res.currency', string='Amount Currency', readonly=True,
+        help="Currency of the 'Amount' field.")
+    amount = fields.Monetary(
+        string='Amount', currency_field='currency_id', readonly=True,
+        help="Amount that is used as base to compute the Cut-off Amount. "
+        "This Amount is in the 'Amount Currency', which may be different "
+        "from the 'Company Currency'.")
+    cutoff_amount = fields.Monetary(
+        string='Cut-off Amount', currency_field='company_currency_id',
+        readonly=True,
+        help="Cut-off Amount without taxes in the Company Currency.")
+    tax_ids = fields.Many2many(
+        'account.tax', column1='cutoff_line_id', column2='tax_id',
+        string='Taxes', readonly=True)
+    tax_line_ids = fields.One2many(
+        'account.cutoff.tax.line', 'parent_id', string='Cut-off Tax Lines',
+        readonly=True)
+
+
+class AccountCutoffTaxLine(models.Model):
+    _name = 'account.cutoff.tax.line'
+    _description = 'Account Cut-off Tax Line'
+
+    parent_id = fields.Many2one(
+        'account.cutoff.line', string='Account Cut-off Line',
+        ondelete='cascade', required=True)
+    tax_id = fields.Many2one('account.tax', string='Tax', required=True)
+    cutoff_account_id = fields.Many2one(
+        'account.account', string='Cut-off Account',
+        domain=[('deprecated', '=', False)], required=True, readonly=True)
+    analytic_account_id = fields.Many2one(
+        'account.analytic.account', string='Analytic Account',
+        domain=[('account_type', '!=', 'closed')], readonly=True)
+    base = fields.Monetary(
+        string='Base', currency_field='currency_id',
+        readonly=True, help="Base Amount in the currency of the PO.")
+    amount = fields.Monetary(
+        string='Tax Amount', currency_field='currency_id',
+        readonly=True, help='Tax Amount in the currency of the PO.')
+    sequence = fields.Integer(string='Sequence', readonly=True)
```

Comment:
The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
    sequence = fields.Integer(readonly=True)
```